tests/unit/test_jobs.py

```diff
@@ -0,0 +1,33 @@
+from catkin_tools.jobs.utils import merge_envs
+
+
+def test_merge_envs_basic():
+    job_env = { 'PATH': '/usr/local/bin:/usr/bin', 'FOO': 'foo' }
+
+    merge_envs(job_env, [
+        { 'PATH': '/usr/local/bin:/bar/baz/bin' },
+        { 'BAR': 'bar' } ])
+
+    # Validate that the known path was not moved from the existing order, and the unfamiliar
+    # path was correctly prepended.
+    assert job_env['PATH'] == '/bar/baz/bin:/usr/local/bin:/usr/bin'
```

Comment:
That doesn't work though, since each dependency will independently include its own copy of the underlay chain. So if you imagine an underlay U containing packages A,B,C, and then a workspace being built which contains updated versions of the three packages, with A' and B' being the isolated resultspaces that have been built. When it comes time to build C (which depends on A and B), what you want in your env path vars is A':B':U or B':A':U, and what you definitely do _not_ want is A':U:B':U or B':U:A':U, since then you'll have one or the other of the updated packages shadowed behind the version in the underlay.

In the real world, I don't think many/any packages are contributing more than one entry to any of the X_PATH vars, and even for those that do, relative order is maintained _within entries that don't overlap with something already present_. Given that, I think what is here is a reasonable compromise in simplicity without getting into an actual constraint satisfaction solver or whatever— particularly if it's constrained to the isolated case, which is a bit off the beaten track no matter how you slice things.